module-system/sov-cli/src/workflows/transactions.rs

```diff
@@ -1,107 +1,129 @@
+        Json: CliFrontEnd<RT>,
+        File: TryInto<RT::CliStringRepr<V>, Error = E1>,
+        Json: TryInto<RT::CliStringRepr<V>, Error = E2>,
+        RT::CliStringRepr<V>: TryInto<RT::Decodable, Error = E3>,
+        RT::Decodable: Serialize,
+        E1: Into<anyhow::Error> + Send + Sync,
+        E2: Into<anyhow::Error> + Send + Sync,
+        E3: Into<anyhow::Error> + Send + Sync,
+    {
         match self {
-            TransactionWorkflow::Generate(subcommand) => {
-                // let TransactionSubcommand { args, inner } = subcommand;
-                let tx: RT::Decodable = subcommand.into();
-                println!("Adding the following transaction to batch:");
-                println!("{}", serde_json::to_string_pretty(&tx)?);
-                wallet_state.unsent_transactions.push(tx);
-            }
-            TransactionWorkflow::Import(subcommand) => {
-                let TransactionSubcommand { args: _, inner } = subcommand;
-                let tx = match inner {
-                    ImportTransaction::FromFile { path } => {
-                        let tx = std::fs::read_to_string(path)?;
-                        serde_json::from_str(&tx)?
-                    }
-                    ImportTransaction::FromString { json } => serde_json::from_str(&json)?,
-                };
-                println!("Adding the following transaction to batch:");
-                println!("{}", serde_json::to_string_pretty(&tx)?);
-                wallet_state.unsent_transactions.push(tx);
-            }
+            TransactionWorkflow::Import(import_workflow) => import_workflow.run(wallet_state),
             TransactionWorkflow::List => {
                 println!("Current batch:");
                 println!(
                     "{}",
                     serde_json::to_string_pretty(&wallet_state.unsent_transactions)?
                 );
+                Ok(())
+            }
+            TransactionWorkflow::Clean => {
+                wallet_state.unsent_transactions.clear();
+                Ok(())
+            }
+            TransactionWorkflow::Remove { index } => {
+                wallet_state.unsent_transactions.remove(index);
+                Ok(())
             }
         }
-
-        Ok(())
     }
 }
+/// An argument passed as path to a file
+#[derive(clap::Parser)]
+pub struct FileArg {
+    /// The path to the file
+    #[arg(long, short)]
+    pub path: String,
+}
 
 #[derive(clap::Subcommand)]
 /// Import a pre-formatted transaction from a JSON file or as a JSON string
-pub enum ImportTransaction {
+pub enum ImportTransaction<Json: Subcommand, File: Subcommand> {
     /// Import a transaction from a JSON file at the provided path
-    #[command(arg_required_else_help(true))]
-    FromFile {
-        /// The expected format of the file contents is: {"module_name": {"call_name": {"field_name": "field_value"}}}
-        path: PathBuf,
-    },
+    #[clap(subcommand)]
+    FromFile(Json),
```

Comment:
Is it possible to use `Box< Subcommand>` for  `FromFile(Json)`, and for ` FromString(File)`? This way, we could eliminate generics from `ImportTransaction`. Currently, we will need to keep adding the generics as we add new subcommands.

Reply:
Great question. I think there should be a way to get rid of these, but it was taking a long time to satisfy the type checker so I wanted to merge this PR to unblock Nikolai first. I'll keep following up on this as I have time.